### ISSUE TITLE
feature: Support for smooth cursor animation under Visual

### ISSUE THREAD

**Issue by CodingOIer** [vscode-neovim/vscode-neovim#2326]
### Did you check the docs?

- [X] I have read all the vscode-neovim docs
- [X] I understand the fundamental limitations of the plugin and believe this request to be reasonable

### Is your feature request related to a problem? Please describe.

I like animations in the editor, even though they may affect performance, but I noticed that the cursor in Visual selection mode does not animate

### Describe the solution you'd like

I hope to apply smooth cursor animation to Visual mode as well

### Describe alternatives you've considered

Added support for smooth cursor animation in Visual mode

### Additional context

https://github.com/user-attachments/assets/cf23003c-e150-4174-9158-e8dfe5eebd6b

**Comment by justinmk** [comment]
I guess it's related to this: https://github.com/vscode-neovim/vscode-neovim/blob/bb0389ac13c5215280cd0f66b32e80e3c916055c/src/viewport_manager.ts#L144C13-L159

**Comment by CodingOIer** [comment]
I love smoothing, I have turned on everything I can, but it still doesn't work

![image](https://github.com/user-attachments/assets/9be9bf67-b3e9-4f22-b201-8a5fd46a6cf6)

**Comment by theol0403** [comment]
Vscode has the "cursor" and the "selection" tightly coupled, ie the cursor position is defined by the selected region. However, vscode treats the cursor position differently than nvim, resulting in an off-by-one error. 

This means in visual mode, the vscode selection is driven by the nvim selection, but the cursor position is wrong so the real vscode cursor is hidden. Then, a fake cursor is rendered using decorations.

TLDR the cursor shown in visual mode is a fake cursor and cannot take advantage of vscode features like cursor animation, and this is required for the visual mode to select the correct text.
